src/Features/CSharp/Portable/Copilot/CSharpCopilotNotImplementedMethodFixProvider.Analyzer.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.MethodImplementation;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Copilot;
+
+internal sealed partial class CSharpCopilotNotImplementedMethodFixProvider
+{
+    private static class DocumentAnalyzer
+    {
+        private const int MaxMethodLength = 1024;
+        private const int ContextLineCount = 10;
```

Comment:
it would be good to make these options.  that way we can override with featur flags to see if it helps/hurts in a/b tests.